GeoInterfaceMakie/src/GeoInterfaceMakie.jl

```diff
@@ -0,0 +1,116 @@
+module GeoInterfaceMakie
+
+using GeoInterface
+import MakieCore as MC
+import GeometryBasics as GB
+import GeoInterface as GI
+import Makie
+
+
+function _plottype(geom)
+    plottype_from_geomtrait(GI.geomtrait(geom))
+end
+function plottype_from_geomtrait(::Union{GI.LineStringTrait, GI.MultiLineStringTrait})
+    MC.Lines
+end
+function plottype_from_geomtrait(::Union{GI.PointTrait, GI.MultiPointTrait})
+    MC.Scatter
+end
+function plottype_from_geomtrait(::Union{GI.PolygonTrait,GI.MultiPolygonTrait, GI.LinearRingTrait})
+    Makie.Poly
+end
+function pttype(geom)
+    if GI.is3d(geom)
+        GB.Point3f
+    else
+        GB.Point2f
+    end
+end
+function points(geom)::Union{Vector{GB.Point2f}, Vector{GB.Point3f}}
+    Pt = pttype(geom)
+    out = Pt[]
+    points!(out, GI.geomtrait(geom), geom)
+end
+@noinline function points!(out, ::GI.AbstractTrait, geom)
+    for pt in GI.getpoint(geom)
+        push!(out, _convert(eltype(out), pt))
+    end
+    out
+end
+@noinline function points!(out, ::GI.PointTrait, pt)
+    push!(out, _convert(eltype(out), pt))
+    out
+end
+function _convert(::Type{GB.Point2f}, pt)
+    x,y = GI.getcoord(pt)
+    GB.Point2f(x,y)
+end
+function _convert(::Type{GB.Point3f}, pt)
+    x,y,z = GI.getcoord(pt)
+    GB.Point3f(x,y,z)
+end
+
+function basicsgeom(geom)
+    t = GI.geomtrait(geom)
+    T = basicsgeomtype(t)
+    GI.convert(T, t, geom)
+end
+
+basicsgeomtype(::GI.PointTrait)           = GB.Point
+basicsgeomtype(::GI.MultiPointTrait)      = GB.MultiPoint
+basicsgeomtype(::GI.PolygonTrait)         = GB.Polygon
+basicsgeomtype(::GI.MultiPolygonTrait)    = GB.MultiPolygon
+basicsgeomtype(::GI.LineStringTrait)      = GB.LineString
+basicsgeomtype(::GI.MultiLineStringTrait) = GB.MultiLineString
+
+function _convert_arguments(::Type{<:Makie.Poly}, geom)::Tuple
+    geob = basicsgeom(geom)::Union{GB.Polygon, GB.MultiPolygon}
+    (geob,)
+end
+function _convert_arguments(::MC.PointBased, geom)::Tuple
+    pts = points(geom)
+    (pts,)
+end
+
+function expr_enable(Geom)
+    quote
+        # import GeoInterfaceMakie.MakieCore
+        # import GeoInterfaceMakie.Makie
+        function $Makie.plottype(geom::$Geom)
```

Comment:
`plottype` is in MakieCore, can we remove the Makie.jl dependency? We can only really use MakieCore in JuliaGeo packages.

Reply:
You are right about this one, however sadly Poly is in Makie and not in MakieCore.

Reply:
Damn. But we can't have everything depend on Makie.jl. 

I'll make an issue over there to get it moved

Reply:
I already made a PR there, no idea if it gets merged or what the obstacles are.

Reply:
Oh thanks. 

I'm not sure why anything returned by `plottype` is not in MakieCore, it kind of defeats the purpose of having a recipe system.